### DISCUSSION TITLE
Ability to view all state variables and their usage

### DISCUSSION THREAD

**Discussion by benaunt** [Budibase/budibase#13382] [Ideas]
It would be beneficial, especially for larger apps if there was a screen to view all variables.

In this screen it should:

- List the variables
- Show all components where the value of the variable is set, allowing the user to click on the component and open up to config where its set
- Show all components where the variable has been references, as above with the ability to click and open the reference
- Ability to provide additional documentation/comments against the variable
- Ability to export the list with all the above data for external documentation.

**Comment by ConorWebb96** [comment]
I really like this idea. 🚀 

I've created a feature request for this you can find it here #13405. As i myself sometimes get lost in larger application and feel it would be rather good to be able to see were state values are being used/set etc.
